src/utils/self_test/tests/Makefile

```diff
@@ -0,0 +1,44 @@
+
```

Comment:
why are we adding makefiles ?

Reply:
When Ethan brought over the configini library/object, he created a simple makefile and tests to verify that nothing was broken while doing the port.  I have kept these in as a sanity check while I was cleaning up the code (a lot to do) and adding new functions.  I kept them in for that purpose so debugging any changes to the file could be kept very local.  These are not necessary along with the utils/self_test/test directory, which is also new.  However, they could be useful if we are to expand the usage of configini.